overlord/snapstate/autorefresh.go

```diff
@@ -237,6 +238,35 @@ func (m *autoRefresh) Ensure() error {
 
 	// do refresh attempt (if needed)
 	if !m.nextRefresh.After(now) {
+		var can bool
+
+		can, err = canRefreshOnMeteredConnection(m.state)
+		if err != nil {
+			return err
+		}
+		if !can {
+			metered, _ := IsOnMeteredConnection()
+			if metered {
+				if now.Sub(lastRefresh) < maxPostponement {
+					var when string
+					switch remaining := int(lastRefresh.Add(maxPostponement).Sub(now).Hours() / 24); remaining {
```

Comment:
this function is already longish, it feels at least this bit could go in a helper